docs: document the Kafka v2 preview release (#5231)
n8n-io/n8n-docs: docs/integrations/builtin/trigger-nodes/n8n-nodes-base.kafkatrigger.md

For usage examples and templates to help you get started, refer to n8n's [Kafka Trigger integrations](https://n8n.io/integrations/kafka-trigger/) page.
{% endhint %}

## Kafka Trigger version 2 preview

{% hint style="info" %}
**Feature availability**

Version 2 of the Kafka Trigger is a preview. Preview features may change in future releases, so avoid relying on them in production workflows.
{% endhint %}

Version 2 of the Kafka Trigger runs on [`@confluentinc/kafka-javascript`](https://github.com/confluentinc/confluent-kafka-javascript), Confluent's supported client built on librdkafka. It replaces kafkajs, the unmaintained library that version 1 uses. The trigger keeps the same fields and credential as version 1, and adds two behaviors:

- It decodes messages compressed with any codec (**GZIP**, **LZ4**, **Snappy**, and **Zstd**), while version 1 only handles uncompressed and GZIP messages.
- It checks the connection when you publish the workflow. If the topic is missing or the consumer can't join its group, publishing fails with a clear error instead of the trigger consuming nothing.

> Version 2 stays off during the preview. Triggers added from the nodes panel stay on version 1.3, and there's no version picker in the editor. To opt in, import a workflow JSON where the Kafka Trigger node has `"typeVersion": 2`. n8n verified the preview on the official n8n Docker images, on both amd64 and arm64. Version 2 includes a native compiled component, so other install methods, such as npm or custom images, aren't verified during the preview.

Reviewer: ```suggestion
Version 2 is disabled by default preview. Triggers added from the nodes panel stay on version 1.3, and there's no version picker in the editor. To opt in, import a workflow JSON where the Kafka Trigger node has `"typeVersion": 2`. n8n has verified the preview on the official n8n Docker images, on both amd64 and arm64. Version 2 includes a native compiled component, so other install methods, such as npm or custom images, aren't verified during the preview.
```
/nitpick
